aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -102,18 +102,19 @@ void rsqrt_kernel_cuda(TensorIteratorBase& iter) {
             const T one = T{1};
             return one / std::sqrt(x);
       }); // rsqrt_string
-      AT_DISPATCH_COMPLEX_TYPES(common_dtype, "rsqrt_cuda", [&]() {
+      AT_DISPATCH_COMPLEX_TYPES_AND(kComplexHalf, common_dtype, "rsqrt_cuda", [&]() {
           jitted_gpu_kernel<
               /*name=*/rsqrt_name,
               /*return_dtype=*/scalar_t,
               /*common_dtype=*/scalar_t,
               /*arity=*/1>(iter, rsqrt_string);
       });
     #else
-      AT_DISPATCH_COMPLEX_TYPES(common_dtype, "rsqrt_cuda", [&]() {
+      AT_DISPATCH_COMPLEX_TYPES_AND(kComplexHalf, common_dtype, "rsqrt_cuda", [&]() {
         gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
+          using opmath_t = at::opmath_type<scalar_t>;
           // In CUDA, ::rsqrt is overloaded for float and at::Half here is implicitly cast to float.
```

Comment:
Is this comment correct? `float` and `half` are not present on this path, right?

Reply:
Hi, @kshitij12345! 
I think no. We are explicitly using `opmath_t` to cast `c10::complex<c10::Half>` into `c10::complex<float>`. I will update this :)
Please let me know if I'm not right. Thanks!